lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -5934,6 +5947,19 @@ void TypeChecker::checkConformancesInContext(DeclContext *dc,
                 "@NSKeyedArchiveLegacy(\"<#class archival name#>\")");
           }
         }
+
+        // If the class declaration doesn't have the
+        // @_staticInitializeObjCMetadata attribute but requires one because
+        // either we complained above, have @NSKeyedArchiveLegacy, or
+        // have generic ancestry, add @_staticInitializeObjCMetadata to indicate
+        // that we need to statically initialize Objective-C metadata.
+        if (!classDecl->getAttrs()
+              .hasAttribute<StaticInitializeObjCMetadataAttr>() &&
+            (kind || hasGenericAncestry(classDecl) ||
+             classDecl->getAttrs().hasAttribute<NSKeyedArchiveLegacyAttr>())) {
```

Comment:
Doesn't NSKeyedArchiveLegacy mean _staticInitializeObjCMetadata is unnecessary? I guess it doesn't harm anything, but…